### PR TITLE
Properties

### PR DESCRIPTION
**Pull request by pictureb00k**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/item/TestItemIdentifierGenerator.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+/**
+ * Generates the unique identifier for test item based
+ * on Base64 encoding and includes information about
+ * name of item's launch, full path of item's parent names,
+ * item name and parameters.
+ *
+ * @author Pavel_Bortnik
+ */
+@Service
+public class TestItemIdentifierGenerator {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/item/TestItemIdentifierGenerator.java]: Let's make it an interface with implementation

File: src/main/java/com/epam/ta/reportportal/core/item/TestItemIdentifierGenerator.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        String forEncoding = prepareForEncoding(testItem);
+        return encoder.encodeToString(forEncoding.getBytes(StandardCharsets.UTF_8));
+    }
+
+    private String prepareForEncoding(TestItem testItem) {
+        String launchName = launchRepository.findNameNumberAndModeById(testItem.getLaunchRef()).getName();
+        List<String> pathNames = getPathNames(testItem.getPath());
+        String itemName = testItem.getName();
+        List<String> parameters = Optional.ofNullable(testItem.getParameters()).orElse(Collections.emptyList());
+        String result = new StringJoiner("").add(launchName).add(pathNames.stream().collect(Collectors.joining())).add(itemName)
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/item/TestItemIdentifierGenerator.java]: let's use comma or smth like that as delimeter

File: src/main/java/com/epam/ta/reportportal/core/item/TestItemIdentifierGenerator.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                .add(parameters.stream().collect(Collectors.joining())).toString();
+        return result.replaceAll("\\s", "");
+    }
+
+    private List<String> getPathNames(List<String> path) {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/item/TestItemIdentifierGenerator.java]: why not streams? Do not allocate array with default size

File: src/main/java/com/epam/ta/reportportal/core/launch/impl/MergeLaunchHandler.java [hunk 4]
```diff
@@ -197,11 +201,13 @@ private void validateMergingLaunches(List<Launch> launches, User user, Project p
...
      * Update test-items of specified launches with new LaunchID
      */
-    private void updateChildrenOfLaunches(String launchId, Set<String> launches, boolean extendDescription) {
+    private void updateChildrenOfLaunches(String launchId, Set<String> launches,
+                                          boolean extendDescription) {
         List<TestItem> testItems = launches.stream().flatMap(id -> {
             Launch launch = launchRepository.findOne(id);
             return testItemRepository.findByLaunch(launch).stream().map(item -> {
                 item.setLaunchRef(launchId);
+                item.setUniqueId(identifierGenerator.generate(item));
```
Comment (pictureb00k) [inline src/main/java/com/epam/ta/reportportal/core/launch/impl/MergeLaunchHandler.java]: Potential bug here. Add project name for base64 and think about if id was set by user, or launch name was changed

File: src/main/java/com/epam/ta/reportportal/core/item/TestItemUniqueIdGenerator.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    private String prepareForEncoding(TestItem testItem) {
+        String launchName = launchRepository.findNameNumberAndModeById(testItem.getLaunchRef()).getName();
+        List<String> pathNames = getPathNames(testItem.getPath());
+        String itemName = testItem.getName();
+        List<String> parameters = Optional.ofNullable(testItem.getParameters()).orElse(Collections.emptyList());
+        String result = new StringJoiner(",").add(launchName).add(pathNames.stream().collect(Collectors.joining())).add(itemName)
+                .add(parameters.stream().collect(Collectors.joining())).toString();
+        return result.replaceAll("\\s", "");
```
Comment (pictureb00k) [inline src/main/java/com/epam/ta/reportportal/core/item/TestItemUniqueIdGenerator.java]: remove

**Pull request merged**